### PR TITLE
Add Polars backend

### LINKED ISSUE #96: Implement polars backend
Polars (almost) completely wraps arrow tables.  Biggest difference is w.r.t it's handling of strings/dictionaries - Since this difference accounts for possibly-significant performance costs in converting to/from arrow, it'd be nice to have direct support for Polars

(I might take a stab at this myself)

### PR DESCRIPTION
**Pull request by kdkavanagh**
Closes #96

### CODE REVIEW

**Comment by tconbeer** [comment]
Thanks for this! I'm away from my computer for a little while but I'll give you a review when I get back

File: src/textual_fastdatatable/backend.py [hunk 1]
```diff
@@ -31,25 +45,35 @@ def create_backend(
...
+        return PolarsBackend.from_dataframe(data, max_rows=max_rows)
+
+    if isinstance(data, Path) or isinstance(data, str):
+        data = Path(data)
+        if data.suffix in [".pqt", ".parquet"]:
+            return ArrowBackend.from_parquet(data, max_rows=max_rows)
+
+        return PolarsBackend.from_file_path(
+            data, max_rows=max_rows, has_header=has_header
+        )
```
Comment (tconbeer) [inline src/textual_fastdatatable/backend.py:60]:
```suggestion
        if pl is not None:
            return PolarsBackend.from_file_path(
                data, max_rows=max_rows, has_header=has_header
            )
```

File: src/textual_fastdatatable/backend.py [hunk 2]
```diff
@@ -387,3 +414,206 @@ def py_str(_ctx: Any, arr: pa.Array) -> str | pa.Array | pa.ChunkedArray:
...
+    ) -> "PolarsBackend":
+        return cls(pl.from_dict(pydict), max_rows=max_rows)
+
+    @classmethod
+    def from_dataframe(
+        cls, frame: pl.DataFrame, max_rows: int | None = None
+    ) -> "PolarsBackend":
+        return cls(frame, max_rows=max_rows)
+
+    def __init__(self, data: pl.DataFrame, max_rows: int | None = None) -> None:
```
Comment (tconbeer) [inline src/textual_fastdatatable/backend.py:449]: Would be good to throw a friendly exception here if Polars isn't installed (`pl is None`)

**Review by tconbeer — CHANGES_REQUESTED** [review]
This looks really good -- I'd just prefer if the dependency on Polars was optional. Also curious how this backend benchmarks vs. arrow?

**Comment by kdkavanagh** [comment]
Updated - I moved the whole class definition under the `try`.  This makes pylint/mypy a bit happier + avoids needing all the `if not none` checks

**Comment by kdkavanagh** [comment]
Benchmark results suggest it may be the fastest impl:

Records |Built-In DataTable | FastDataTable (Arrow from Parquet) | FastDataTable (Arrow from Records) | FastDataTable (Polars from Parquet)
--------|--------|--------|--------|--------
lap_times_100.parquet |   0.008s /   1.304s |   0.006s /   1.406s |   0.007s /   1.441s |   0.023s /   1.145s
lap_times_1000.parquet |   0.050s /   1.749s |   0.007s /   1.933s |   0.005s /   1.866s |   0.005s /   1.098s
lap_times_10000.parquet |   0.339s /   1.985s |   0.006s /   1.909s |   0.009s /   1.862s |   0.005s /   1.107s
lap_times_100000.parquet |   3.171s /   4.755s |   0.012s /   1.895s |   0.053s /   1.932s |   0.013s /   1.096s
lap_times_538121.parquet |  17.126s /  18.921s |   0.025s /   1.933s |   0.272s /   2.138s |   0.033s /   1.131s
wide_10000.parquet |   2.216s /   5.913s |   0.009s /   2.362s |   0.023s /   2.291s |   0.007s /   1.086s
wide_100000.parquet |  22.291s /  26.005s |   0.023s /   2.300s |   0.184s /   2.552s |   0.011s /   1.091s

File: pyproject.toml [hunk 3]
```diff
@@ -18,9 +18,13 @@ build-backend = "poetry.core.masonry.api"
 python = "^3.8"
 textual = ">=0.41.0"
 pyarrow = ">=7.0.0"
+polars = { version = ">=0.20.0", optional = true }
```
Comment (tconbeer) [inline pyproject.toml:21]: You'll also need to add `polars` as a test dependency in the `tool.poetry.group.test.dependencies` below.

Comment (tconbeer) [reply pyproject.toml:21]: (and also as an `additional_dependency` for the `mypy` hook in `pre-commit-config.yaml`)

File: src/textual_fastdatatable/__init__.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 from textual_fastdatatable.backend import (
     ArrowBackend,
+    PolarsBackend,
```
Comment (tconbeer) [inline src/textual_fastdatatable/__init__.py:3]: This only exists if `_HAS_POLARS`

**Review by tconbeer — CHANGES_REQUESTED** [review]

**Comment by tconbeer** [comment]
The benchmark results are exciting -- the optional import stuff still needs some work, though.

**Pull request merged**